developer/src/common/web/test-helpers/verifyCompilerMessagesObject.ts

```diff
@@ -0,0 +1,66 @@
+import { CompilerErrorSeverity, compilerErrorSeverityName } from '@keymanapp/common-types';
+import {assert, expect} from 'chai';
+
+//
+// The purpose of these tests is to ensure that our slightly WET constant
+// definitions for compiler errors are consistent in their usage:
+//
+//   * that names are consistent between the message functions and constants
+//   * that constants have unique codes
+//   * that constants have the right error mask for their type
+//
+// This means we don't have to worry about DRYing out the constants, which would
+// probably require us to have a preprocessing step, which has its own
+// negatives.
+//
+// These are used in each of the test-messages.ts units to verify the
+// CompilerMessages classes
+
+const toTitleCase = (s: string) => s.substring(0, 1).toUpperCase() + s.substring(1).toLowerCase();
+
+export function verifyCompilerMessagesObject(source: Record<string,any>) {
+  let keys = Object.keys(source);
+
+  const m = source as Record<string,any>;
+
+  let codes: number[] = [];
+
+  for(let key of keys) {
+
+    // Verify each object member matches the pattern we expect
+
+    if(typeof m[key] == 'function') {
+      const o = /^(Info|Hint|Warn|Error|Fatal)_([A-Za-z0-9_]+)$/.exec(key);
```

Comment:
It makes me wonder if this should really be derived from the output of `compilerErrorSeverityName()`.